### DISCUSSION TITLE
Facing Issue In Rendering Unicode Fonts

### DISCUSSION THREAD

**Discussion by Anish-Someashwara** [protectwise/troika#302] [Q&A]
Hey @lojjic firstly thanks for this amazing work, it’s working great.

For past few days, I was working to add Unicode Fonts in my threejs project using troika-three-text.
I am using troika-three-text for some native languages of India, I have tried it for unicode fonts it’s working, but for some character and ligature its not rendering properly and not looking same as the original text. I have attached some picture Reference below:

**Example: 1**
**Font Used:** Devanagiri
**Original Text:** दिव्या
**Troika Rendered Text:**
![Hindi-Pblm](https://github.com/protectwise/troika/assets/154493929/df59c9c1-84fc-4e03-9e28-8c5207626f4e)

**Example: 2**
**Font Used:** Gurmukhi
**Original Text:** ਵਿਦਿਆਰਥੀ ਕਾਲ ਮਨੁੱਖ ਦੇ ਭਾਵੀ ਜੀਵਨ ਦੀ ਬੁਨਿਆਦ ਹੁੰਦਾ ਹੈ, ਕਿਉਂਕਿ ਨੀਂਹ ਜਿੰਨੀ ਮਜ਼ਬੂਤ ਅਤੇ ਪੱਕੀ ਹੋਵੇਗੀ, ਉਸ ਉੱਤੇ ਉੱਨਾ ਹੀ ਮਜ਼ਬੂਤ ਮਹਲ ਉਸਾਰਿਆ ਜਾ ਸਕਦਾ ਹੈ।
**Troika Rendered Text:**
![pnbi-Pblm](https://github.com/protectwise/troika/assets/154493929/f10b97d3-5d4b-4080-a671-b4b464c82796)

So, These are the issues I am facing, what can I do for this?
Do I need to specify the ligature sequences, will it work?

And one more thing i want to ask is, can we customize the glyphs, Like I am making a game for kids, so I want to make the font design to look a little cartoon and comic style.

Thanks In Advance

**Comment by lojjic** [comment]
Thanks, unfortunately there are many details of [Indic text shaping](https://github.com/n8willis/opentype-shaping-documents/blob/master/opentype-shaping-indic-general.md) that I have not yet implemented. It's quite complex and I don't know when I'll be able to get to it. Sorry I don't have a better answer for you.

**Reply by Anish-Someashwara** [reply]
Thanks @lojjic for your response. 
Can you please suggest me that what other possibilities I have to achieve the same goal.
And What I can do to contribute on the same problem for troika-three-text?

**Comment by lojjic** [comment]
Created https://github.com/protectwise/troika/issues/303 to track this.

**Reply by amitrajput1992** [reply]
@lojjic Is there a workaround for this?
